### PR TITLE
Redundant computations in Interval::contains

### PR DESCRIPTION
**Pull request by abhinavagarwal07**

### CODE REVIEW

**Comment by nishnik** [comment]
+1 to merge if tests pass.

File: symengine/sets.cpp [hunk 1]
```diff
@@ -94,13 +94,17 @@ RCP<const Boolean> Interval::contains(const RCP<const Basic> &a) const
 {
     if (not is_a_Number(*a))
         return make_rcp<Contains>(a, rcp_from_this_cast<const Set>());
-    if ((eq(*start_, *a) and left_open_) or (eq(*end_, *a) and right_open_))
-        return boolean(false);
-    if (eq(*start_, *a) or eq(*end_, *a))
+    if (eq(*start_, *a)) {
+        if (left_open_)
+            return boolean(false);
         return boolean(true);
```
Comment (isuruf) [inline symengine/sets.cpp]: You can do `return boolean(not left_open_)` instead of above 3 lines.

File: symengine/sets.cpp [hunk 2]
```diff
@@ -94,13 +94,17 @@ RCP<const Boolean> Interval::contains(const RCP<const Basic> &a) const
...
-    if (eq(*min({end_, a}), *end_))
-        return boolean(false);
-    if (eq(*max({start_, a}), *start_))
+    }
+    if (eq(*end_, *a)) {
+        if (right_open_)
+            return boolean(false);
+        return boolean(true);
```
Comment (isuruf) [inline symengine/sets.cpp]: Same as above

**Comment by isuruf** [comment]
Left a comment. Otherwise +1

File: symengine/sets.cpp (hunk 1 above)
Comment (abhinavagarwal07) [inline symengine/sets.cpp]: Yeah that will be better

**Pull request merged**